projector-plugin/src/main/kotlin/org/jetbrains/projector/plugin/ui/ProjectorStatusWidget.kt

```diff
@@ -144,18 +151,42 @@ class ProjectorStatusWidget(private val myStatusBar: StatusBar)
       val ctx = DataManager.getInstance().getDataContext(myStatusBar as Component)
       val event = AnActionEvent.createFromAnAction(action, null, "", ctx)
       action.actionPerformed(event)
-    } else {
+    }
+    else {
       logger.error("Unable to get action with ID = $actionId")
     }
   }
 
   override fun propertyChange(event: PropertyChangeEvent?) {
     event?.let {
-      clients = event.newValue as Int
-      SwingUtilities.invokeLater { update() }
+      when (event.propertyName) {
+        "clientsCount" -> {
+          clients = event.newValue as Int
+          SwingUtilities.invokeLater { update() }
+        }
+
+        "macLocalConnection" -> {
```

Comment:
Could we maybe introduce named constants for these strings?

Reply:
You a right. I've encapsulated the server -> plugin communication in ServerMessages module